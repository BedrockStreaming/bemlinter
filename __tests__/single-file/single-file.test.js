const bemlinter = require('./../../src/bemlinter');

it('should log a lint without error', (done) => {
  bemlinter(`${__dirname}/alright.scss`)
    .then(logs => {
      expect(logs).toMatchSnapshot();
      done();
    });
});

it('should log a lint with a leak error', (done) => {
  bemlinter(`${__dirname}/leak.scss`)
    .then(logs => {
      expect(logs).toMatchSnapshot();
      done();
    });
});

it('should log a lint with 6 syntax error', (done) => {
  bemlinter(`${__dirname}/syntax.scss`)
    .then(logs => {
      expect(logs).toMatchSnapshot();
      done();
    });
});

it('should log a lint without lower case error', (done) => {
  bemlinter(`${__dirname}/syntax.scss`, [], {checkLowerCase: false})
    .then(logs => {
      expect(logs).toMatchSnapshot();
      done();
    });
});