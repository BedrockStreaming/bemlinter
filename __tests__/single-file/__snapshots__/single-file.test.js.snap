exports[`test should log one block linted with 6 syntax error 1`] = `
Object {
  "syntax": Object {
    "error": Array [
      Object {
        "filePath": "./__tests__/single-file/syntax.scss",
        "line": 26,
        "message": "\".syntax---big\" modifier should have only 2 dashes.",
      },
      Object {
        "filePath": "./__tests__/single-file/syntax.scss",
        "line": 31,
        "message": "\".syntax--big--light\" should have a single modifier.",
      },
      Object {
        "filePath": "./__tests__/single-file/syntax.scss",
        "line": 37,
        "message": "\".syntax__Close\" should be in lower case.",
      },
      Object {
        "filePath": "./__tests__/single-file/syntax.scss",
        "line": 45,
        "message": "\".syntax--big__close\" represents an element of a modifier, it should be cut in 2 classes.",
      },
      Object {
        "filePath": "./__tests__/single-file/syntax.scss",
        "line": 54,
        "message": "\".syntax__header__logo\" should have a single depth of element.",
      },
      Object {
        "filePath": "./__tests__/single-file/syntax.scss",
        "line": 58,
        "message": "\".syntax___footer\" element should have only 2 underscores.",
      },
    ],
    "info": Array [
      Object {
        "filePath": "./__tests__/single-file/syntax.scss",
        "line": null,
        "message": "Linting finish",
      },
    ],
  },
}
`;

exports[`test should log one block linted with a leak error 1`] = `
Object {
  "leak": Object {
    "error": Array [
      Object {
        "filePath": "./__tests__/single-file/leak.scss",
        "line": 15,
        "message": "\".external-component\" is incoherent with the file name.",
      },
    ],
    "info": Array [
      Object {
        "filePath": "./__tests__/single-file/leak.scss",
        "line": null,
        "message": "Linting finish",
      },
    ],
  },
}
`;

exports[`test should log one block linted without error 1`] = `
Object {
  "alright": Object {
    "info": Array [
      Object {
        "filePath": "./__tests__/single-file/alright.scss",
        "line": null,
        "message": "Linting finish",
      },
    ],
  },
}
`;
